src/main/java/com/spotify/futures/CompletableFutures.java

```diff
@@ -237,4 +243,40 @@ private CompletableFutures() {
         .thenApply(ignored ->
                        function.apply(af.join(), bf.join(), cf.join(), df.join(), ef.join()));
   }
+
+  /**
+   * Periodically poll an external resource until it returns a non-empty result.
+   *
+   * The polling task should return Optional.empty() until it becomes available, and then
+   * Optional.of(result).
+   *
+   * If the polling task throws an exception or returns null, that will cause the result future to
+   * complete exceptionally.
+   *
+   * @param pollingTask the polling task.
+   * @param frequency the frequency to run the polling task at.
+   * @param executorService the executor service to schedule the polling task on.
+   * @return a future completing to the result of the polling task once that becomes available.
+   */
+  public static <T> CompletableFuture<T> poll(
+      final Supplier<Optional<T>> pollingTask,
+      final Duration frequency,
```

Comment:
It might be an idea to leave the scheduling of the task up to the caller, by for example returning something that `implements Runnable` and has `public CompletableFuture<T> future()` on it.  Not sure if that's possible in a nice way though.